src/target/linux.rs

```diff
@@ -221,6 +306,192 @@ impl LinuxTarget {
 
         Ok(tasks)
     }
+
+    #[cfg(target_arch = "x86_64")]
+    pub fn set_hardware_breakpoint(
+        &mut self,
+        breakpoint: HardwareBreakpoint,
+    ) -> Result<usize, Box<dyn std::error::Error>> {
+        #[cfg(target_arch = "x86_64")]
+        {
+            let index = if let Some(empty) = self.find_empty_watchpoint() {
+                empty
+            } else {
+                return Err(Box::new(HardwareBreakpointError::NoEmptyWatchpoint));
+            };
+
+            let rw_bits: u64 = match breakpoint.typ {
+                HardwareBreakpointType::Execute => 0b00,
+                HardwareBreakpointType::Read => 0b11,
+                HardwareBreakpointType::ReadWrite => 0b11,
+                HardwareBreakpointType::Write => 0b01,
+            } << 16 + index * 4;
+
+            let size_bits: u64 = (breakpoint.size as u64) << (18 + index * 4);
+
+            let enable_bit: u64 = 1 << (2 * index);
+
+            let bit_mask: u64 = (0b11 << (2 * index)) | (0b1111 << (16 + 4 * index));
+
+            let mut dr7: u64 =
+                self.ptrace_peekuser((*DEBUG_REG_OFFSET + 7 * 8) as *mut libc::c_void)? as u64;
+
+            // Check if hardware watchpoint is already used
+            if dr7 & (1 << (2 * index)) != 0 {
+                // Panic for now
+                panic!("Invalid debug register state")
+            }
+
+            dr7 = (dr7 & !bit_mask) | (enable_bit | rw_bits | size_bits);
+
+            #[allow(deprecated)]
+            unsafe {
+                //Have to use deprecated function because of no alternative for PTRACE_POKEUSER
+                ptrace::ptrace(
+                    ptrace::Request::PTRACE_POKEUSER,
+                    self.pid,
+                    (*DEBUG_REG_OFFSET + index * 8) as *mut libc::c_void,
+                    breakpoint.addr as *mut libc::c_void,
+                )?;
+                ptrace::ptrace(
+                    ptrace::Request::PTRACE_POKEUSER,
+                    self.pid,
+                    (*DEBUG_REG_OFFSET + 7 * 8) as *mut libc::c_void,
+                    dr7 as *mut libc::c_void,
+                )?;
+                ptrace::ptrace(
+                    ptrace::Request::PTRACE_POKEUSER,
+                    self.pid,
+                    (*DEBUG_REG_OFFSET + 6 * 8) as *mut libc::c_void,
+                    0 as *mut libc::c_void,
+                )?;
+            }
+
+            self.hardware_breakpoints[index] = Some(breakpoint);
+
+            Ok(index)
+        }
+        #[cfg(not(target_arch = "x86_64"))]
+        Err(Box::new(HardwareBreakpointError::UnsupportedPlatform))
```

Comment:
We can probably replace
```rust
#[cfg(target_arch = "x86_64")]
{
    let index = ...;
}
#[cfg(not(target_arch = "x86_64"))]
Err(Box::new(HardwareBreakpointError::UnsupportedPlatform))
```
with
```rust
if !cfg!(target_arch = "x86_64")  {
    return Err(Box::new(HardwareBreakpointError::UnsupportedPlatform));
}
let index = ...;
```
Since it will be translated into something like
```rust
if true { return Err(...) }
```
on non-x86_64 platforms, the compiler should be smart enough to eleminate unused code.

Reply:
`DEBUG_REG_OFFSET` only exists on x86_64.